### PR TITLE
Updated the upload template for processing the v3 study doc.

### LINKED ISSUE #1050: Temporary edits to current submission documentation/forms
### Inspiration

We would like a bare-bones functioning submission interface ASAP, so that may occur before the submission form is refactored based on discussion #1040.  In the meantime, the submission instructions are currently outdated.

### Description

Update both the upload.html template and the google submission form to reflect the current process, which is basically:

1. Upload all the peak annotation files you have (noting you don't need/shouldn't try to manually create/should not upload, a study doc) to get back a study doc template that has been autopopulated with your samples
2. Manually flesh the study doc (providing tips/pointers)
3. Upload the study doc only on the same build-a-submission page to get back a validation report
4. Try to fix simple errors and retry step 3 as much as you are willing to do (or skip to step 5).  Note that validation is optional, but will speed up the loading of your submission
5. Submit the google form

### Alternatives

None

### Dependencies

This issue cannot be started until the completion of the following issue(s)/
pull request(s):

- #issue_number_1
- #pull_request_1

### Comment

None

-----

### Assumptions

1. List of assumptions that the code will not explicitly address/check
4. E.g. We will assume input is correct (explaining why there is no validation)

### Limitations

1. A list of things this work will specifically not do
2. E.g. This feature will only handle the most frequent use case X

### Affected Components

- change: File path or DB table ...
- add: Environment variable or server setting
- delete: External executable or cron job

### Requirements

- [ ] 1. List of numbered conditions to be met for the feature
- [ ] 2. E.g. Every column/row must display a value, i.e. cannot be empty
- [ ] 3. Numbers for reference & checkboxes for progress tracking

#### Interface Change description

None provided

#### Code Change Description

None provided

#### Tests

- [ ] 1. A description of at least one test for each requirement above.
- [ ] 2. E.g. Test for req 2 that there's an exception when display value is ''
- [ ] 3. Numbers for reference & checkboxes for progress tracking

### PR DESCRIPTION
**Pull request by hepcat72**
## Summary Change Description

See affected issue(s).

## Affected Issues/Pull Requests

- Resolves #1050
- Merges into PR #1085
- Next PR: #1129

## Review Notes

See comments in-line.

## Checklist

This pull request will be merged once the following requirements are met.  The
author and/or reviewers should uncheck any unmet requirements:

- Review requirements
  - Minimum approvals: 1 
  - No changes requested
  - All blocking issues resolved by reviewers
  - Specific reviewers: @__add_username_here__
    
  - Review period: 2 days 
- Associated issue/pull request requirements:
  
  - [x] All requirements in affected issues marked "resolved" are satisfied
  - [x] All required pull requests are merged *(or none)*
- Basic requirements
  
  - [x] [All linters pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#linting)
  - [x] [All tests pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#quality-control)
  - [x] All conflicts resolved
- Overhead requirements
  
  - [x] [New/changed method tests implemented/updated *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#test-implementation)
  - [ ] [Updated *Unreleased* section of `changelog.md` *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/changelog.md)
  - [x] [Migrations created & committed *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#migration-process)

### CODE REVIEW

File: DataRepo/templates/upload.html [hunk 1]
```diff
@@ -115,88 +340,336 @@ <h4 id="animal-sample-accucor-data-submission-process">Data Submission Process</
...
+                    You will get back a new (or updated) <span class="tooltip-hint" text-tooltip="Animal/Sample excel file">study doc</span> copy that is automatically downloaded.  If any issues
                     are detected, a status report will be added to the page.
                 </li>
                 <li>
-                    Move on to the next section, but note that after you flesh out your study doc, you can at any point,
-                    come back to this page and submit your study file and peak annotation files to automatically
+                    Move on to the next section, but note that after you flesh out your <span class="tooltip-hint" text-tooltip="Animal/Sample excel file">study doc</span>, you can at any point,
+                    come back to this page and submit your <span class="tooltip-hint" text-tooltip="Animal/Sample excel file">study doc</span> and <span class="tooltip-hint" text-tooltip="E.g. AccuCor excel file">peak annotation files</span> to automatically
                     add more samples, get a status report detailing any issues uncovered in your data, and also
```
Comment (lparsons) [inline DataRepo/templates/upload.html:393]: The previous section of documentation seemed to discourage this due to the possibility of a gateway time out error. Suggesting they users come back and submit peak annotation files along with a study doc seems at odds with the previous advice "Generally, _do not_ select any existing study doc".

**Review by lparsons — APPROVED** [review]
I've added some comments, questions, and suggestions inline. I hesitate to mark anything "blocking" since I know much of this is already changed later, so I'll mark as "Approve" and let you decide how to best handle each comment.

Having the instructions separate from the form makes this harder to follow for users. They have to read each section, then click away to actually submit the files, etc. It would be preferable to include the instructions for submitting peak annotation files along with the form to submit peak annotation files.

The javascript is following an existing pattern, so it makes sense to continue it this way. However,  I'd like to make few points for consideration on future templates:
1. Separating javascript into a separate file and including would be preferable to long inline scripts. It makes it easier to lint, read, etc. when it's split apart, imho.
2. There is a lot of code to handle collapsing sections, and it seems like an existing bootstrap mechanism could be used for that.  Something like https://getbootstrap.com/docs/4.0/components/collapse/
3. There are cookies to remember the state of each element. In addition to adding a lot of code to maintain, I'd probably prefer that the state not be remembered on successive page views, but that functionality can be discussed during testing.

File: DataRepo/templates/upload.html [hunk 2]
```diff
@@ -115,88 +340,336 @@ <h4 id="animal-sample-accucor-data-submission-process">Data Submission Process</
     <hr />
     <small class="text-muted">
         Feel free to submit data as you generate it.  Your study does not have to be complete for submission.  You can
-        iteratively submit accucor/isocorr files with their metadata described in the study doc (aka animal/sample
-        sheet), as you finish them.  Each one can be an additional separate submission to the same study.<br>
+        iteratively submit <span class="tooltip-hint" text-tooltip="E.g. AccuCor excel file">peak annotation files</span> with their metadata described in the <span class="tooltip-hint" text-tooltip="Animal/Sample excel file">study doc</span>, as you finish them.  Each one can be an additional separate submission to the same study.<br>
```
[the remark this answers is not in the document]
Comment (hepcat72) [reply DataRepo/templates/upload.html:343]: Yes.  The Animals sheet's Study column does a database lookup using the study name, as if it was a foreign key.  It's the same mechanism we use in all of the loaders' model relations.  Animal name, Sample name, Tissue name, Treatment name, etc.

File: DataRepo/templates/upload.html [hunk 3]
```diff
@@ -115,88 +340,336 @@ <h4 id="animal-sample-accucor-data-submission-process">Data Submission Process</
...
+                            &nbsp;<a data-bs-toggle="collapse" data-bs-target="#autofill-details"><img id="autofill-details-button" src="{% static 'images/plus.png' %}" alt="more" width="18"></a>
+                            <div id="autofill-details" class="collapse">
+                                <ul>
+                                    <li>
+                                        Note, you can submit a partial study now and add to a previously submitted study later.
+                                    </li>
+                                    <li>
+                                        You may include:
+                                        <ul>
+                                            <li>Accucor <em>(excel or the Corrected sheet as tsv or csv)</em></li>
```
[the remark this answers is not in the document]
Comment (hepcat72) [reply DataRepo/templates/upload.html:366]: I'm going to leave this as-is.  This file is deleted later on, though this change should be made to where this was copied.

File: DataRepo/templates/upload.html [hunk 4]
```diff
@@ -115,88 +340,336 @@ <h4 id="animal-sample-accucor-data-submission-process">Data Submission Process</
...
+                                            <li>Isocorr <em>(excel)</em></li>
```
[the remark this answers is not in the document]
Comment (hepcat72) [reply DataRepo/templates/upload.html:367]: We should keep an eye out to make this, the previous, and any other changes in the places where these were copied before this file was deleted.

File: DataRepo/templates/upload.html [hunk 5]
```diff
@@ -115,88 +340,336 @@ <h4 id="animal-sample-accucor-data-submission-process">Data Submission Process</
...
+                        <li>
+                            Generally, <em>do not</em> select any existing <span class="tooltip-hint" text-tooltip="Animal/Sample excel file">study doc</span>.
+                            &nbsp;<a data-bs-toggle="collapse" data-bs-target="#nostudydoc-details"><img id="nostudydoc-details-button" src="{% static 'images/plus.png' %}" alt="more" width="18"></a>
+                            <div id="nostudydoc-details" class="collapse">
+                                <ul>
+                                    <li>You can submit an existing <span class="tooltip-hint" text-tooltip="Animal/Sample excel file">study doc</span> to add samples to it from the supplied <span class="tooltip-hint" text-tooltip="E.g. AccuCor excel file">peak annotation files</span>, but <em>only</em> if it has not been manually edited.  Otherwise, you would likely encounter a gateway timeout error, as this functionality is currently under development.</li>
+                                    <li><em>Supplying peak annotations and a <span class="tooltip-hint" text-tooltip="Animal/Sample excel file">study doc</span> together is currently under development.</em></li>
```
[the remark this answers is not in the document]
Comment (hepcat72) [reply DataRepo/templates/upload.html:380]: This goes away in the new flow version with the tabs.

File: DataRepo/templates/upload.html (hunk 1 above)
Comment (hepcat72) [reply DataRepo/templates/upload.html:393]: I edited all these for brevity and these caveats in the latest version.  This was before I realized phase 1 would be the breakup of the process into 4 pages, so this goes away in the latest version.

File: DataRepo/templates/upload.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (lparsons) [reply DataRepo/templates/upload.html:343]: We should make that clear in our instructions to users, it wasn't entirely clear to me the way it's written here. Just a note for future documentation efforts.

[the remark this answers is not in the document]
Comment (hepcat72) [reply DataRepo/templates/upload.html:343]: OK.  Keep that in mind when you review these instructions on the submission page in the next PRs.  I had made some edits after the copy (and delete of this file), so maybe it's clearer.  If not, we can make an edit on that file.

**Comment by hepcat72** [comment]
> Having the instructions separate from the form makes this harder to follow for users. They have to read each section, then click away to actually submit the files, etc. It would be preferable to include the instructions for submitting peak annotation files along with the form to submit peak annotation files.

Yes, as the phase 1 plan lays out, these instructions moved to the second tab in later PRs.  This was a temporary update, partially borne of a misunderstanding about whether or not there was a quick update before phase 1.  At this point however, phase 1 is fully implemented and this PR is mostly moot.

> The javascript is following an existing pattern, so it makes sense to continue it this way. However, I'd like to make few points for consideration on future templates:
> 
> 1. Separating javascript into a separate file and including would be preferable to long inline scripts. It makes it easier to lint, read, etc. when it's split apart, imho.

I agree, and you'll see that that's what I did in the later PRs.

> 2. There is a lot of code to handle collapsing sections, and it seems like an existing bootstrap mechanism could be used for that.  Something like https://getbootstrap.com/docs/4.0/components/collapse/

This does use the bootstrap mechanism to which you refer.  If there's a specific change regarding how that was implemented, let me know.

> 3. There are cookies to remember the state of each element. In addition to adding a lot of code to maintain, I'd probably prefer that the state not be remembered on successive page views, but that functionality can be discussed during testing.

I initially didn't have cookies remembering the collapse states.  When the user had to go back and forth between the instructions and the form, it was annoying to get back to where you were in the instructions.  Now that (in later PRs) the instructions are on the page, this may no longer be necessary.

That said however, I suspect that code volume you're referring to is mostly attributable to [the bootstrap code to which you referred above](https://getbootstrap.com/docs/4.0/components/collapse/), so I'm not sure removing the cookie code will address this concern.

**Pull request merged**